lib/prompts/autocomplete.js

```diff
@@ -0,0 +1,203 @@
+/**
+ * `list` type prompt
+ */
+
+var _ = require("lodash");
+var util = require("util");
+var chalk = require("chalk");
+var figures = require("figures");
+var Base = require("./base");
+var Choices = require("../objects/choices");
+var observe = require("../utils/events");
+var utils = require("../utils/readline");
+var Paginator = require("../utils/paginator");
+
+
+/**
+ * Module exports
+ */
+
+module.exports = Prompt;
+
+
+/**
+ * Constructor
+ */
+
+function Prompt() {
+  Base.apply(this, arguments);
+
+  if (!this.opt.choices) {
+    this.throwParamError("choices");
+  }
+
+  this.currentChoices = [];
+
+  this.firstRender = true;
+  this.selected = 0;
+
+  var def = this.opt.default;
```

Comment:
Not sure what to do with default for this prompt, since we have a external unknown at runtime resultset. If you want a default, maybe have the service return that choice if no match?